src/transport/assembly_dg.hh

```diff
@@ -25,137 +25,434 @@
 
-    virtual void set_boundary_conditions(DHCellAccessor cell) = 0;
+    /// Method prepares object before assemblation (e.g. balance, ...).
+    virtual void begin(unsigned int subst_idx) {}
 
-    virtual void prepare_initial_condition(DHCellAccessor cell) = 0;
+    /// Method finishes object after assemblation (e.g. balance, ...).
+    virtual void end(unsigned int subst_idx) {}
 };
 
 
 /**
  * Auxiliary container class for Finite element and related objects of given dimension.
  */
 template <unsigned int dim, class Model>
-class AssemblyDG : public AssemblyDGBase
+class MassAssemblyDG : public AssemblyBase<dim>
 {
 public:
     typedef typename TransportDG<Model>::EqData EqDataDG;
 
     /// Constructor.
-    AssemblyDG(std::shared_ptr<EqDataDG> data, TransportDG<Model> &model)
-    : fe_(make_shared< FE_P_disc<dim> >(data->dg_order)), fe_low_(make_shared< FE_P_disc<dim-1> >(data->dg_order)),
-      fe_rt_(new FE_RT0<dim>), fe_rt_low_(new FE_RT0<dim-1>),
-      quad_(new QGauss(dim, 2*data->dg_order)),
-	  quad_low_(new QGauss(dim-1, 2*data->dg_order)),
-      model_(model), data_(data){
-
-        fv_rt_.initialize(*quad_, *fe_rt_, update_values | update_gradients | update_quadrature_points);
-        fe_values_.initialize(*quad_, *fe_, update_values | update_gradients | update_JxW_values | update_quadrature_points);
-        fe_values_side_.initialize(*quad_low_, *fe_, update_values | update_gradients | update_side_JxW_values | update_normal_vectors | update_quadrature_points);
-        fsv_rt_.initialize(*quad_low_, *fe_rt_, update_values | update_quadrature_points);
-
-        if (dim>1) {
-            fv_rt_vb_.initialize(*quad_low_, *fe_rt_low_, update_values | update_quadrature_points);
-            fe_values_vb_.initialize(*quad_low_, *fe_low_,
-                    update_values | update_gradients | update_JxW_values | update_quadrature_points);
-        }
-        ndofs_ = fe_->n_dofs();
-        qsize_ = quad_->size();
-        qsize_lower_dim_ = quad_low_->size();
-        dof_indices_.resize(ndofs_);
-        loc_dof_indices_.resize(ndofs_);
-        side_dof_indices_vb_.resize(2*ndofs_);
+    MassAssemblyDG(std::shared_ptr<EqDataDG> data)
+    : model_(nullptr), data_(data), fe_values_(nullptr) {
+        quad_ = new QGauss(dim, 2*data_->dg_order);
+        quad_low_ = new QGauss(dim-1, 2*data_->dg_order);
     }
 
     /// Destructor.
-    ~AssemblyDG() {
-        delete fe_rt_;
-        delete fe_rt_low_;
+    ~MassAssemblyDG() {
         delete quad_;
         delete quad_low_;
+
+        if (fe_values_!=nullptr) delete fe_values_;
+
+    }
+
+    void create_integrals(std::shared_ptr<EvalPoints> eval_points) {
+        bulk_integral_ = eval_points->add_bulk<dim>(*quad_);
+        //edge_integral_ = eval_points->add_edge<dim>(*quad_low_);
+        //if (dim>1) coupling_integral_ = eval_points->add_coupling<dim>(*quad_low_);
+        //boundary_integral_ = eval_points->add_boundary<dim>(*quad_low_);
```

Comment:
Remove dependency in GenericAssembly::add_xyz_integral(), delete unused integral objects. Same in other descendants of AssemblyBase.